site/src/layouts/DesignerSelector.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Link from 'gatsby-link'
+import styled from 'emotion/react'
+import cx from 'classnames'
+import { monofont, sansfont, baseUl } from './emotion-base'
+
+const Container = styled.div`
+  position: fixed;
+  top: 120px;
+  right: 0;
+  padding: 16px 8px;
+  border: 2px solid #000;
+  width: 128px;
+  background: #fff;
+
+  transform: translateX(100px);
+  transition: transform 0.2s ease;
+
+  &.active {
+    transform: none;
+  }
+`
+
+const DesignerList = styled.ul`
+  composes: ${baseUl}, ${monofont};
+  font-size: 14px;
+
+  & li {
+    margin: 0;
+    padding: 0;
+    line-height: 1.25;
+
+    &:not(:last-child) {
+      margin-bottom: 12px;
+    }
+
+    &.active a {
+      background: #ff0;
+      text-decoration: underline;
+    }
+
+    & a {
+      color: inherit;
+      text-decoration: inherit;
+
+      &:hover,
+      &:focus {
+        text-decoration: underline;
+      }
+    }
+  }
+`
+
+class DesignerSelector extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isActive: true,
+      lastActivationTime: Date.now(),
+    }
+  }
+
+  componentDidMount() {
+    // hide self after initial peek
+    setTimeout(() => {
+      this.setState({ isActive: false })
+    }, 1200)
+  }
+
+  onMouseEnter() {
+    this.setState({
+      isActive: true,
+      lastActivationTime: Date.now(),
+    })
+  }
+
+  onMouseLeave() {
+    // before hiding, allow a buffer for user to re-enter the container
+    clearTimeout(this.leaveTimeout)
+    this.leaveTimeout = setTimeout(() => {
+      if (Date.now() - this.state.lastActivationTime > 300) {
+        this.setState({ isActive: false })
+      }
+    }, 300)
+  }
+
+  render() {
+    const { currentDesignerSlug } = this.props
+    const { isActive } = this.state
+
+    // TODO: remove temporary designers
+    const designers = [].concat(this.props.designers)
```

Comment:
Curious what the `[].concat` is for here?

Reply:
If its a cast won't the propTypes take care of that?

Reply:
it is just a quick array copy of the passed in prop, since I am doing that stupid multiplication to test the layout. Will remove once real data is entered.